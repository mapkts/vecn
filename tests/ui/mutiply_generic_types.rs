#[vecn::vector]
struct TupleStruct1<T, U>((T, U), (T, U), (T, U));

#[vecn::vector]
struct NamedStruct1<T, U> {
    x: (T, U),
    y: (T, U),
}

#[vecn::vector]
struct TupleStruct2<T, U>(&(T, U), &(T, U), &(T, U));

#[vecn::vector]
struct NamedStruct2<T, U> {
    x: &(T, U),
    y: &(T, U),
}

fn main() {}
