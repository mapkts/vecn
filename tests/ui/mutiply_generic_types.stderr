error: expected at most one generic type, but found others
 --> $DIR/mutiply_generic_types.rs:2:26
  |
2 | struct TupleStruct1<T, U>((T, U), (T, U), (T, U));
  |                          ^^^^^^^^^^^^^^^^^^^^^^^^

error: expected at most one generic type, but found others
 --> $DIR/mutiply_generic_types.rs:5:27
  |
5 |   struct NamedStruct1<T, U> {
  |  ___________________________^
6 | |     x: (T, U),
7 | |     y: (T, U),
8 | | }
  | |_^

error: expected at most one generic type, but found others
  --> $DIR/mutiply_generic_types.rs:11:26
   |
11 | struct TupleStruct2<T, U>(&(T, U), &(T, U), &(T, U));
   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected at most one generic type, but found others
  --> $DIR/mutiply_generic_types.rs:14:27
   |
14 |   struct NamedStruct2<T, U> {
   |  ___________________________^
15 | |     x: &(T, U),
16 | |     y: &(T, U),
17 | | }
   | |_^
